src/neo/SmartContract/Native/Oracle/OracleContract.cs

```diff
@@ -0,0 +1,186 @@
+#pragma warning disable IDE0051
+
+using Neo.Cryptography;
+using Neo.Cryptography.ECC;
+using Neo.Ledger;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract.Manifest;
+using Neo.VM.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+
+namespace Neo.SmartContract.Native.Oracle
+{
+    public sealed partial class OracleContract : NativeContract
+    {
+        private const int MaxUrlLength = 256;
+        private const int MaxFilterLength = 128;
+        private const int MaxCallbackLength = 32;
+        private const int MaxUserDataLength = 512;
+
+        private const byte Prefix_NodeList = 8;
+        private const byte Prefix_RequestId = 9;
+        private const byte Prefix_Request = 7;
+        private const byte Prefix_IdList = 6;
+
+        private const long OracleRequestPrice = 0_50000000;
+
+        public override int Id => -4;
+        public override string Name => "Oracle";
+
+        internal OracleContract()
+        {
+            Manifest.Features = ContractFeatures.HasStorage;
+        }
+
+        [ContractMethod(0, CallFlags.AllowModifyStates)]
+        private void Finish(ApplicationEngine engine)
+        {
+            Transaction tx = (Transaction)engine.ScriptContainer;
+            OracleResponse response = tx.GetAttribute<OracleResponse>();
```

Comment:
```suggestion
            OracleResponse response = tx.GetAttribute<OracleResponse>();
            if (response == null) throw new ArgumentException("Oracle response was not found");
```

Reply:
What's the difference? If `response` is `null`, it will throw exception in the next line because of `response.Id`.

Reply:
Yes, it's not a fix, its for debugging purpose now that we have the fault exception it's good to provide the specific error.